src/apps/service/apps/agave/listings.clj

```diff
@@ -1,24 +1,58 @@
 (ns apps.service.apps.agave.listings
-  (:use [apps.service.util :only [sort-apps apply-offset apply-limit uuid?]]
+  (:use [apps.service.util :only [sort-apps apply-offset apply-limit format-job-stats uuid?]]
+        [apps.util.conversions :only [remove-nil-vals]]
         [slingshot.slingshot :only [try+]])
   (:require [clojure.tools.logging :as log]
             [clojure-commons.error-codes :as ce :refer [clj-http-error?]]
-            [apps.persistence.app-metadata :as ap]))
+            [apps.persistence.app-metadata :as ap]
+            [apps.persistence.jobs :as jobs-db]))
+
+(defn- add-agave-job-stats
+  [{:keys [id] :as app} admin?]
+  (log/spy id)
+  (let [job-stats     (if admin?
+                        (jobs-db/get-job-stats id)
+                        (jobs-db/get-public-job-stats id))
+        default-stats (if admin?
+                        {:job_count 0
+                         :job_count_failed 0
+                         :job_count_completed 0}
+                        {:job_count_completed 0})]
+    (merge app default-stats job-stats)))
+
+(defn- add-app-listing-job-stats
+  [apps admin?]
+  (update apps :apps (partial map (comp remove-nil-vals #(add-agave-job-stats % admin?)))))
+
+(defn- format-app-listing-job-stats
+  [apps admin?]
+  (update apps :apps (partial map #(format-job-stats % admin?))))
+
+(defn get-app-details
+  [agave app-id admin?]
+  (-> (.getAppDetails agave app-id)
+      (add-agave-job-stats admin?)
+      (format-job-stats admin?)
+      remove-nil-vals))
 
 (defn list-apps
   [agave category-id params]
   (-> (.listApps agave)
+      (add-app-listing-job-stats false)
```

Comment:
Unfortunately we have to query the job stats of every Agave app in the initial listing, before applying limits and offsets, in case the request wishes to sort the results by one of the job stats fields.